fetch.bs

```diff
@@ -6710,6 +6748,364 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+         is <a>allowed to use</a> the <a>policy-controlled feature</a>
+         {{PermissionPolicy/"deferred-fetch-minimal"}}, then increment <var>quota</var> by
+         <a>minimal subframe deferred-fetch quota</a>.
+        </ol>
+      </ol>
+
+     <li>
+      <p><a for=list>For each</a> <a>deferred fetch record</a> <var>deferredRecord</var> of
+      <var>otherDocument</var>'s <a for=fetch>fetch group</a>'s
+      <a for="fetch group">deferred fetch records</a>:</p>
+      <ol>
+       <li><p>Let <var>requestLength</var> be the <a>total request length</a> of
+       <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>.
+
+       <li><p>Decrement <var>quota</var> by <var>requestLength</var>.
+
+       <li><p>If <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>'s
+       <a for=request>URL</a>'s <a for=url>origin</a> is <a>same origin</a> with <var>origin</var>,
+       then decrement <var>quotaForOrigin</var> by <var>requestLength</var>.
+      </ol>
+    </ol>
+
+   <li>
+    <p>Otherwise, if <var>otherNavigable</var> is not a <a for=/>top-level traversable</a>, and its
+    <a for=navigable>parent</a>'s <a>active document</a> and <var>document</var>
+    <a>share deferred-fetch quota</a>, then decrement <var>quota</var> by
+    <var>otherNavigable</var>'s <a>navigable container</a>'s
+    <a>subframe reserved deferred-fetch quota</a>.
+
+    <p class=note>This is a cross-origin/cross-agent subframe, so we check if it has reserved some
+    quota which was delegated via {{PermissionsPolicy}}.
+  </ol>
+ <li><p>If <var>quota</var> is less than zero, than return zero.
+
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then
+ return <var>quota</var>.
+
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
+<var>container</var> given an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called when the container document and the document that initiated the
+navigation (the "source document") <a>share deferred-fetch quota</a>. It potentially
+reserves either 64kb or 8kb of quota for the frame, if it doesn't <a>share deferred-fetch quota</a>
+with its parent and the permissions policy allow. It is not observable to the cotnainer document
+whether the reserved quota was used in practice.
+This algorithm assumes that the container's document might delegate quota to the navigated frame,
+and the reserved quota would only apply in that case, and would be ignored if it ends up being
+shared.
+
+<ol>
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionPolicy/"deferred-fetch"}}, <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>container</var>'s <a>container document</a> is equal or greater than
```

Comment:
```
and the available deferred-fetch quota for container’s container document is equal or greater than optional subframe deferred-fetch quota, t
```

Given the following examples

```
root document (a.com) -> <div> -> frame_a (a.com) -> <div> -> frame_b (a.com)
```

* What is `container's container document` when the `navigable container` is `frame_a`? Is it root document?
* What is `container's container document` when the `navigable container` is `frame_b`? Is it `frame_a`?

Reply:
1. yes
2. yes
3. Yea, I need to clean that up